src/ui/common/src/components/workflows/storageSelector.tsx

```diff
@@ -0,0 +1,59 @@
+import Box from '@mui/material/Box';
+import FormControl from '@mui/material/FormControl';
+import MenuItem from '@mui/material/MenuItem';
+import Select from '@mui/material/Select';
+import Typography from '@mui/material/Typography';
+import React from 'react';
+import { useSelector } from 'react-redux';
+
+import { RootState } from '../../stores/store';
+import { StorageType, StorageTypeNames } from '../../utils/storage';
+
+export const StorageSelector: React.FC = () => {
+  const workflow = useSelector((state: RootState) => state.workflowReducer);
+  const dag = workflow.selectedDag;
+  let selected = 'file';
+  if (dag) {
+    selected = dag.storage_config.type;
+  }
+
+  const getMenuItems = () => {
+    return (
+      Object.values(StorageType) as Array<
+        typeof StorageType[keyof typeof StorageType]
+      >
+    ).map((r, _) => {
```

Comment:
I think you can just omit the second parameter of .map here.

Try doing:
```typescript
.map((r) => ...
```

Reply:
Could you also use a more descriptive name here please? Something like storageType should suffice.